<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>

<title></title>

<script type="text/javascript">
window.onload = function() {
  var imgs = document.getElementsByTagName('img'), i, img;
  for (i = 0; i < imgs.length; i++) {
    img = imgs[i];
    // center an image if it is the only element of its parent
    if (img.parentElement.childElementCount === 1)
      img.parentElement.style.textAlign = 'center';
  }
};
</script>

<!-- Styles for R syntax highlighter -->
<style type="text/css">
   pre .operator,
   pre .paren {
     color: rgb(104, 118, 135)
   }

   pre .literal {
     color: #990073
   }

   pre .number {
     color: #099;
   }

   pre .comment {
     color: #998;
     font-style: italic
   }

   pre .keyword {
     color: #900;
     font-weight: bold
   }

   pre .identifier {
     color: rgb(0, 0, 0);
   }

   pre .string {
     color: #d14;
   }
</style>

<!-- R syntax highlighter -->
<script type="text/javascript">
var hljs=new function(){function m(p){return p.replace(/&/gm,"&amp;").replace(/</gm,"&lt;")}function f(r,q,p){return RegExp(q,"m"+(r.cI?"i":"")+(p?"g":""))}function b(r){for(var p=0;p<r.childNodes.length;p++){var q=r.childNodes[p];if(q.nodeName=="CODE"){return q}if(!(q.nodeType==3&&q.nodeValue.match(/\s+/))){break}}}function h(t,s){var p="";for(var r=0;r<t.childNodes.length;r++){if(t.childNodes[r].nodeType==3){var q=t.childNodes[r].nodeValue;if(s){q=q.replace(/\n/g,"")}p+=q}else{if(t.childNodes[r].nodeName=="BR"){p+="\n"}else{p+=h(t.childNodes[r])}}}if(/MSIE [678]/.test(navigator.userAgent)){p=p.replace(/\r/g,"\n")}return p}function a(s){var r=s.className.split(/\s+/);r=r.concat(s.parentNode.className.split(/\s+/));for(var q=0;q<r.length;q++){var p=r[q].replace(/^language-/,"");if(e[p]){return p}}}function c(q){var p=[];(function(s,t){for(var r=0;r<s.childNodes.length;r++){if(s.childNodes[r].nodeType==3){t+=s.childNodes[r].nodeValue.length}else{if(s.childNodes[r].nodeName=="BR"){t+=1}else{if(s.childNodes[r].nodeType==1){p.push({event:"start",offset:t,node:s.childNodes[r]});t=arguments.callee(s.childNodes[r],t);p.push({event:"stop",offset:t,node:s.childNodes[r]})}}}}return t})(q,0);return p}function k(y,w,x){var q=0;var z="";var s=[];function u(){if(y.length&&w.length){if(y[0].offset!=w[0].offset){return(y[0].offset<w[0].offset)?y:w}else{return w[0].event=="start"?y:w}}else{return y.length?y:w}}function t(D){var A="<"+D.nodeName.toLowerCase();for(var B=0;B<D.attributes.length;B++){var C=D.attributes[B];A+=" "+C.nodeName.toLowerCase();if(C.value!==undefined&&C.value!==false&&C.value!==null){A+='="'+m(C.value)+'"'}}return A+">"}while(y.length||w.length){var v=u().splice(0,1)[0];z+=m(x.substr(q,v.offset-q));q=v.offset;if(v.event=="start"){z+=t(v.node);s.push(v.node)}else{if(v.event=="stop"){var p,r=s.length;do{r--;p=s[r];z+=("</"+p.nodeName.toLowerCase()+">")}while(p!=v.node);s.splice(r,1);while(r<s.length){z+=t(s[r]);r++}}}}return z+m(x.substr(q))}function j(){function q(x,y,v){if(x.compiled){return}var u;var s=[];if(x.k){x.lR=f(y,x.l||hljs.IR,true);for(var w in x.k){if(!x.k.hasOwnProperty(w)){continue}if(x.k[w] instanceof Object){u=x.k[w]}else{u=x.k;w="keyword"}for(var r in u){if(!u.hasOwnProperty(r)){continue}x.k[r]=[w,u[r]];s.push(r)}}}if(!v){if(x.bWK){x.b="\\b("+s.join("|")+")\\s"}x.bR=f(y,x.b?x.b:"\\B|\\b");if(!x.e&&!x.eW){x.e="\\B|\\b"}if(x.e){x.eR=f(y,x.e)}}if(x.i){x.iR=f(y,x.i)}if(x.r===undefined){x.r=1}if(!x.c){x.c=[]}x.compiled=true;for(var t=0;t<x.c.length;t++){if(x.c[t]=="self"){x.c[t]=x}q(x.c[t],y,false)}if(x.starts){q(x.starts,y,false)}}for(var p in e){if(!e.hasOwnProperty(p)){continue}q(e[p].dM,e[p],true)}}function d(B,C){if(!j.called){j();j.called=true}function q(r,M){for(var L=0;L<M.c.length;L++){if((M.c[L].bR.exec(r)||[null])[0]==r){return M.c[L]}}}function v(L,r){if(D[L].e&&D[L].eR.test(r)){return 1}if(D[L].eW){var M=v(L-1,r);return M?M+1:0}return 0}function w(r,L){return L.i&&L.iR.test(r)}function K(N,O){var M=[];for(var L=0;L<N.c.length;L++){M.push(N.c[L].b)}var r=D.length-1;do{if(D[r].e){M.push(D[r].e)}r--}while(D[r+1].eW);if(N.i){M.push(N.i)}return f(O,M.join("|"),true)}function p(M,L){var N=D[D.length-1];if(!N.t){N.t=K(N,E)}N.t.lastIndex=L;var r=N.t.exec(M);return r?[M.substr(L,r.index-L),r[0],false]:[M.substr(L),"",true]}function z(N,r){var L=E.cI?r[0].toLowerCase():r[0];var M=N.k[L];if(M&&M instanceof Array){return M}return false}function F(L,P){L=m(L);if(!P.k){return L}var r="";var O=0;P.lR.lastIndex=0;var M=P.lR.exec(L);while(M){r+=L.substr(O,M.index-O);var N=z(P,M);if(N){x+=N[1];r+='<span class="'+N[0]+'">'+M[0]+"</span>"}else{r+=M[0]}O=P.lR.lastIndex;M=P.lR.exec(L)}return r+L.substr(O,L.length-O)}function J(L,M){if(M.sL&&e[M.sL]){var r=d(M.sL,L);x+=r.keyword_count;return r.value}else{return F(L,M)}}function I(M,r){var L=M.cN?'<span class="'+M.cN+'">':"";if(M.rB){y+=L;M.buffer=""}else{if(M.eB){y+=m(r)+L;M.buffer=""}else{y+=L;M.buffer=r}}D.push(M);A+=M.r}function G(N,M,Q){var R=D[D.length-1];if(Q){y+=J(R.buffer+N,R);return false}var P=q(M,R);if(P){y+=J(R.buffer+N,R);I(P,M);return P.rB}var L=v(D.length-1,M);if(L){var O=R.cN?"</span>":"";if(R.rE){y+=J(R.buffer+N,R)+O}else{if(R.eE){y+=J(R.buffer+N,R)+O+m(M)}else{y+=J(R.buffer+N+M,R)+O}}while(L>1){O=D[D.length-2].cN?"</span>":"";y+=O;L--;D.length--}var r=D[D.length-1];D.length--;D[D.length-1].buffer="";if(r.starts){I(r.starts,"")}return R.rE}if(w(M,R)){throw"Illegal"}}var E=e[B];var D=[E.dM];var A=0;var x=0;var y="";try{var s,u=0;E.dM.buffer="";do{s=p(C,u);var t=G(s[0],s[1],s[2]);u+=s[0].length;if(!t){u+=s[1].length}}while(!s[2]);if(D.length>1){throw"Illegal"}return{r:A,keyword_count:x,value:y}}catch(H){if(H=="Illegal"){return{r:0,keyword_count:0,value:m(C)}}else{throw H}}}function g(t){var p={keyword_count:0,r:0,value:m(t)};var r=p;for(var q in e){if(!e.hasOwnProperty(q)){continue}var s=d(q,t);s.language=q;if(s.keyword_count+s.r>r.keyword_count+r.r){r=s}if(s.keyword_count+s.r>p.keyword_count+p.r){r=p;p=s}}if(r.language){p.second_best=r}return p}function i(r,q,p){if(q){r=r.replace(/^((<[^>]+>|\t)+)/gm,function(t,w,v,u){return w.replace(/\t/g,q)})}if(p){r=r.replace(/\n/g,"<br>")}return r}function n(t,w,r){var x=h(t,r);var v=a(t);var y,s;if(v){y=d(v,x)}else{return}var q=c(t);if(q.length){s=document.createElement("pre");s.innerHTML=y.value;y.value=k(q,c(s),x)}y.value=i(y.value,w,r);var u=t.className;if(!u.match("(\\s|^)(language-)?"+v+"(\\s|$)")){u=u?(u+" "+v):v}if(/MSIE [678]/.test(navigator.userAgent)&&t.tagName=="CODE"&&t.parentNode.tagName=="PRE"){s=t.parentNode;var p=document.createElement("div");p.innerHTML="<pre><code>"+y.value+"</code></pre>";t=p.firstChild.firstChild;p.firstChild.cN=s.cN;s.parentNode.replaceChild(p.firstChild,s)}else{t.innerHTML=y.value}t.className=u;t.result={language:v,kw:y.keyword_count,re:y.r};if(y.second_best){t.second_best={language:y.second_best.language,kw:y.second_best.keyword_count,re:y.second_best.r}}}function o(){if(o.called){return}o.called=true;var r=document.getElementsByTagName("pre");for(var p=0;p<r.length;p++){var q=b(r[p]);if(q){n(q,hljs.tabReplace)}}}function l(){if(window.addEventListener){window.addEventListener("DOMContentLoaded",o,false);window.addEventListener("load",o,false)}else{if(window.attachEvent){window.attachEvent("onload",o)}else{window.onload=o}}}var e={};this.LANGUAGES=e;this.highlight=d;this.highlightAuto=g;this.fixMarkup=i;this.highlightBlock=n;this.initHighlighting=o;this.initHighlightingOnLoad=l;this.IR="[a-zA-Z][a-zA-Z0-9_]*";this.UIR="[a-zA-Z_][a-zA-Z0-9_]*";this.NR="\\b\\d+(\\.\\d+)?";this.CNR="\\b(0[xX][a-fA-F0-9]+|(\\d+(\\.\\d*)?|\\.\\d+)([eE][-+]?\\d+)?)";this.BNR="\\b(0b[01]+)";this.RSR="!|!=|!==|%|%=|&|&&|&=|\\*|\\*=|\\+|\\+=|,|\\.|-|-=|/|/=|:|;|<|<<|<<=|<=|=|==|===|>|>=|>>|>>=|>>>|>>>=|\\?|\\[|\\{|\\(|\\^|\\^=|\\||\\|=|\\|\\||~";this.ER="(?![\\s\\S])";this.BE={b:"\\\\.",r:0};this.ASM={cN:"string",b:"'",e:"'",i:"\\n",c:[this.BE],r:0};this.QSM={cN:"string",b:'"',e:'"',i:"\\n",c:[this.BE],r:0};this.CLCM={cN:"comment",b:"//",e:"$"};this.CBLCLM={cN:"comment",b:"/\\*",e:"\\*/"};this.HCM={cN:"comment",b:"#",e:"$"};this.NM={cN:"number",b:this.NR,r:0};this.CNM={cN:"number",b:this.CNR,r:0};this.BNM={cN:"number",b:this.BNR,r:0};this.inherit=function(r,s){var p={};for(var q in r){p[q]=r[q]}if(s){for(var q in s){p[q]=s[q]}}return p}}();hljs.LANGUAGES.cpp=function(){var a={keyword:{"false":1,"int":1,"float":1,"while":1,"private":1,"char":1,"catch":1,"export":1,virtual:1,operator:2,sizeof:2,dynamic_cast:2,typedef:2,const_cast:2,"const":1,struct:1,"for":1,static_cast:2,union:1,namespace:1,unsigned:1,"long":1,"throw":1,"volatile":2,"static":1,"protected":1,bool:1,template:1,mutable:1,"if":1,"public":1,friend:2,"do":1,"return":1,"goto":1,auto:1,"void":2,"enum":1,"else":1,"break":1,"new":1,extern:1,using:1,"true":1,"class":1,asm:1,"case":1,typeid:1,"short":1,reinterpret_cast:2,"default":1,"double":1,register:1,explicit:1,signed:1,typename:1,"try":1,"this":1,"switch":1,"continue":1,wchar_t:1,inline:1,"delete":1,alignof:1,char16_t:1,char32_t:1,constexpr:1,decltype:1,noexcept:1,nullptr:1,static_assert:1,thread_local:1,restrict:1,_Bool:1,complex:1},built_in:{std:1,string:1,cin:1,cout:1,cerr:1,clog:1,stringstream:1,istringstream:1,ostringstream:1,auto_ptr:1,deque:1,list:1,queue:1,stack:1,vector:1,map:1,set:1,bitset:1,multiset:1,multimap:1,unordered_set:1,unordered_map:1,unordered_multiset:1,unordered_multimap:1,array:1,shared_ptr:1}};return{dM:{k:a,i:"</",c:[hljs.CLCM,hljs.CBLCLM,hljs.QSM,{cN:"string",b:"'\\\\?.",e:"'",i:"."},{cN:"number",b:"\\b(\\d+(\\.\\d*)?|\\.\\d+)(u|U|l|L|ul|UL|f|F)"},hljs.CNM,{cN:"preprocessor",b:"#",e:"$"},{cN:"stl_container",b:"\\b(deque|list|queue|stack|vector|map|set|bitset|multiset|multimap|unordered_map|unordered_set|unordered_multiset|unordered_multimap|array)\\s*<",e:">",k:a,r:10,c:["self"]}]}}}();hljs.LANGUAGES.r={dM:{c:[hljs.HCM,{cN:"number",b:"\\b0[xX][0-9a-fA-F]+[Li]?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+(?:[eE][+\\-]?\\d*)?L\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\b\\d+\\.(?!\\d)(?:i\\b)?",e:hljs.IMMEDIATE_RE,r:1},{cN:"number",b:"\\b\\d+(?:\\.\\d*)?(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"number",b:"\\.\\d+(?:[eE][+\\-]?\\d*)?i?\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"keyword",b:"(?:tryCatch|library|setGeneric|setGroupGeneric)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\.",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\.\\.\\d+(?![\\w.])",e:hljs.IMMEDIATE_RE,r:10},{cN:"keyword",b:"\\b(?:function)",e:hljs.IMMEDIATE_RE,r:2},{cN:"keyword",b:"(?:if|in|break|next|repeat|else|for|return|switch|while|try|stop|warning|require|attach|detach|source|setMethod|setClass)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"literal",b:"(?:NA|NA_integer_|NA_real_|NA_character_|NA_complex_)\\b",e:hljs.IMMEDIATE_RE,r:10},{cN:"literal",b:"(?:NULL|TRUE|FALSE|T|F|Inf|NaN)\\b",e:hljs.IMMEDIATE_RE,r:1},{cN:"identifier",b:"[a-zA-Z.][a-zA-Z0-9._]*\\b",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"<\\-(?!\\s*\\d)",e:hljs.IMMEDIATE_RE,r:2},{cN:"operator",b:"\\->|<\\-",e:hljs.IMMEDIATE_RE,r:1},{cN:"operator",b:"%%|~",e:hljs.IMMEDIATE_RE},{cN:"operator",b:">=|<=|==|!=|\\|\\||&&|=|\\+|\\-|\\*|/|\\^|>|<|!|&|\\||\\$|:",e:hljs.IMMEDIATE_RE,r:0},{cN:"operator",b:"%",e:"%",i:"\\n",r:1},{cN:"identifier",b:"`",e:"`",r:0},{cN:"string",b:'"',e:'"',c:[hljs.BE],r:0},{cN:"string",b:"'",e:"'",c:[hljs.BE],r:0},{cN:"paren",b:"[[({\\])}]",e:hljs.IMMEDIATE_RE,r:0}]}};
hljs.initHighlightingOnLoad();
</script>



<style type="text/css">
body, td {
   font-family: sans-serif;
   background-color: white;
   font-size: 13px;
}

body {
  max-width: 800px;
  margin: auto;
  padding: 1em;
  line-height: 20px;
}

tt, code, pre {
   font-family: 'DejaVu Sans Mono', 'Droid Sans Mono', 'Lucida Console', Consolas, Monaco, monospace;
}

h1 {
   font-size:2.2em;
}

h2 {
   font-size:1.8em;
}

h3 {
   font-size:1.4em;
}

h4 {
   font-size:1.0em;
}

h5 {
   font-size:0.9em;
}

h6 {
   font-size:0.8em;
}

a:visited {
   color: rgb(50%, 0%, 50%);
}

pre, img {
  max-width: 100%;
}
pre {
  overflow-x: auto;
}
pre code {
   display: block; padding: 0.5em;
}

code {
  font-size: 92%;
  border: 1px solid #ccc;
}

code[class] {
  background-color: #F8F8F8;
}

table, td, th {
  border: none;
}

blockquote {
   color:#666666;
   margin:0;
   padding-left: 1em;
   border-left: 0.5em #EEE solid;
}

hr {
   height: 0px;
   border-bottom: none;
   border-top-width: thin;
   border-top-style: dotted;
   border-top-color: #999999;
}

@media print {
   * {
      background: transparent !important;
      color: black !important;
      filter:none !important;
      -ms-filter: none !important;
   }

   body {
      font-size:12pt;
      max-width:100%;
   }

   a, a:visited {
      text-decoration: underline;
   }

   hr {
      visibility: hidden;
      page-break-before: always;
   }

   pre, blockquote {
      padding-right: 1em;
      page-break-inside: avoid;
   }

   tr, img {
      page-break-inside: avoid;
   }

   img {
      max-width: 100% !important;
   }

   @page :left {
      margin: 15mm 20mm 15mm 10mm;
   }

   @page :right {
      margin: 15mm 10mm 15mm 20mm;
   }

   p, h2, h3 {
      orphans: 3; widows: 3;
   }

   h2, h3 {
      page-break-after: avoid;
   }
}
</style>



</head>

<body>
<pre><code class="r">source(&quot;../code/functions.R&quot;)

# set these
# email &lt;- &quot;mark.robinson@mls.uzh.ch&quot;
# orcid &lt;- &quot;0000-0002-3048-5518&quot;
# scholar_id &lt;- &quot;XPfrRQEAAAAJ&quot;
# pubmed_search &lt;- &quot;Robinson Mark D[au]&quot;
# pmid_remove &lt;- c(&quot;28824762&quot;, &quot;26295592&quot;, &quot;23288288&quot;, &quot;21516278&quot;, &quot;18830830&quot;,
#                  &quot;18025499&quot;, &quot;15786672&quot;, &quot;15779224&quot;, &quot;15322224&quot;, &quot;30039500&quot;,
#                  &quot;31925006&quot;)
# pmid_add &lt;- c(&quot;11743205&quot;, &quot;15761153&quot;, &quot;23857251&quot;, &quot;26493315&quot;,
#               &quot;31178352&quot;, &quot;18772890&quot;, &quot;18573797&quot;)

# email &lt;- &quot;wong@immunology.uzh.ch&quot;
# orcid &lt;- &quot;0000-0003-3155-1211&quot;
# pubmed_search &lt;- &quot;wong, w wei-lynn[author] OR Wong, Wendy W-L[Author] OR Wong, Wendy Wei-Lynn[Author] OR (gerondakis s and wong l)&quot;
# scholar_id &lt;- &quot;JynlX_8AAAAJ&quot;
# pmid_remove &lt;- NULL
# pmid_add &lt;- NULL

email &lt;- &quot;mering@imls.uzh.ch&quot;
orcid &lt;- &quot;0000-0001-7734-9102&quot;
pubmed_search &lt;- &quot;von Mering c[au]&quot;
scholar_id &lt;- &quot;Av-VeeEAAAAJ&quot;
pmid_remove &lt;- &quot;18814862&quot;
pmid_add &lt;- NULL


pm_file &lt;- paste0(&quot;PUBMED_&quot;, gsub(&quot;.uzh.ch&quot;,&quot;&quot;, email), &quot;.txt&quot;)
orcid_file &lt;- paste0(&quot;ORCID_&quot;, gsub(&quot;.uzh.ch&quot;,&quot;&quot;, email), &quot;.txt&quot;)
scholar_file &lt;- paste0(&quot;SCHOLAR_&quot;, gsub(&quot;.uzh.ch&quot;,&quot;&quot;, email), &quot;.txt&quot;)
bibtex_file &lt;- paste0(&quot;BIBTEX_FOR_ORCID_&quot;, gsub(&quot;.uzh.ch&quot;,&quot;&quot;, email), &quot;.bib&quot;)

df_pubmed &lt;- retrieve_from_entrez(pubmed_search)
dim(df_pubmed)
</code></pre>

<pre><code>## [1] 115   6
</code></pre>

<pre><code class="r">write.table(df_pubmed, pm_file, sep=&quot;\t&quot;, quote=FALSE, row.names = FALSE)


df_orcid &lt;- retrieve_from_orcid(orcid)
dim(df_orcid)
</code></pre>

<pre><code>## [1] 111   4
</code></pre>

<pre><code class="r"># In ORCID, not in PubMed (mostly preprints?)
setdiff(df_orcid$doi, df_pubmed$doi)
</code></pre>

<pre><code>## [1] &quot;10.1093/nar/gkv1310&quot;        &quot;10.1002/9783527619368.ch32&quot; &quot;10.1002/pmic.201290014&quot;     &quot;10.1093/database/baw100&quot;   
## [5] &quot;10.1093/nar/gky1131&quot;        &quot;10.1101/040097&quot;             &quot;10.1101/126953&quot;             &quot;10.1142/9789812838780_0008&quot;
## [9] &quot;10.1186/s12859-019-2828-z&quot;
</code></pre>

<pre><code class="r"># In Pubmed, not in ORCID
(to_update &lt;- na.omit(setdiff(df_pubmed$doi, df_orcid$doi)))
</code></pre>

<pre><code>##  [1] &quot;10.1038/s41587-020-0434-2&quot;         &quot;10.1038/s41586-020-1965-x&quot;         &quot;10.1038/s41467-020-14367-0&quot;       
##  [4] &quot;10.1172/jci123263&quot;                 &quot;10.1016/j.molcel.2019.04.021&quot;      &quot;10.1016/j.molcel.2019.03.041&quot;     
##  [7] &quot;10.1038/nrd.2018.52&quot;               &quot;10.1016/j.ijantimicag.2017.03.017&quot; &quot;10.1186/s13059-015-0807-x&quot;        
## [10] &quot;10.1093/bioinformatics/btg398&quot;     &quot;10.1038/nature750&quot;                 &quot;10.1016/s0896-6273(00)00046-5&quot;    
## [13] &quot;10.1016/s0960-9822(00)80054-8&quot;
</code></pre>

<pre><code class="r">df_pubmed[df_pubmed$doi %in% to_update,] %&gt;% select(-authors,-pmid)
</code></pre>

<pre><code>##          pubyear
## 32123383    2020
## 32025015    2020
## 32024854    2020
## 31107248    2019
## 31101498    2019
## 31047794    2019
## 29567993    2018
## 28689869    2017
## 26490954    2015
## 14734319    2004
## 12000970    2002
## 10985358    2000
## 10574767    1999
##                                                                                                                                      title
## 32123383                                                                     ChromID identifies the protein interactome at chromatin marks
## 32025015                                                              Analyses of non-coding somatic drivers in 2,658 cancer whole genomes
## 32024854                                                               Pathway and network analysis of more than 2500 whole cancer genomes
## 31107248                              Protein tyrosine phosphatase non-receptor type 22 modulates colitis in a microbiota-dependent manner
## 31101498 Analysis of the Human Kinome and Phosphatome by Mass Cytometry Reveals Overexpression-Induced Effects on Cancer-Related Signaling
## 31047794                                 Cross-Regulation between TDP-43 and Paraspeckles Promotes Pluripotency-Differentiation Transition
## 29567993                                                                          Unexplored therapeutic opportunities in the human genome
## 28689869                                                         Effects of oral antibiotics and isotretinoin on the murine gut microbiota
## 26490954                          Erratum to: gespeR: a statistical model for deconvoluting off-target-confounded RNA interference screens
## 14734319                                  The Helmholtz Network for Bioinformatics: an integrative web portal for bioinformatics resources
## 12000970                                                   Comparative assessment of large-scale data sets of protein-protein interactions
## 10985358                     Prion protein devoid of the octapeptide repeat region restores susceptibility to scrapie in PrP knockout mice
## 10574767                                                             Distinct and regulated activities of human Gli proteins in Drosophila
##                          journal                               doi
## 32123383          Nat Biotechnol         10.1038/s41587-020-0434-2
## 32025015                  Nature         10.1038/s41586-020-1965-x
## 32024854              Nat Commun        10.1038/s41467-020-14367-0
## 31107248           J Clin Invest                 10.1172/jci123263
## 31101498                Mol Cell      10.1016/j.molcel.2019.04.021
## 31047794                Mol Cell      10.1016/j.molcel.2019.03.041
## 29567993     Nat Rev Drug Discov               10.1038/nrd.2018.52
## 28689869 Int J Antimicrob Agents 10.1016/j.ijantimicag.2017.03.017
## 26490954             Genome Biol         10.1186/s13059-015-0807-x
## 14734319          Bioinformatics     10.1093/bioinformatics/btg398
## 12000970                  Nature                 10.1038/nature750
## 10985358                  Neuron     10.1016/s0896-6273(00)00046-5
## 10574767               Curr Biol     10.1016/s0960-9822(00)80054-8
</code></pre>

<pre><code class="r"># write missing entries to bibtex file
bibtex_from_doi &lt;- GetBibEntryWithDOI(to_update)
# toBiblatex(bibtex_from_doi)
writeLines(toBiblatex(bibtex_from_doi), bibtex_file)


df_scholar &lt;- retrieve_from_scholar(scholar_id)
dim(df_scholar)
</code></pre>

<pre><code>## [1] 136   8
</code></pre>

<pre><code class="r">m &lt;- match(toupper(df_scholar$title), toupper(df_orcid$title))
df_scholar$doi &lt;- df_orcid$doi[m]

# non-matching titles
w &lt;- is.na(df_scholar$doi)
# df_scholar$title[w]

# bit of faffing to find what is in Google Scholar, not in ORCID
scores &lt;- calcScore(df_scholar$title[w], df_orcid$title)
top_score &lt;- apply(scores$dist, 1, max)
which_top_score &lt;- apply(scores$dist, 1, which.max)

scholar_to_orcid_matches &lt;- data.frame(top_score, scholar_title=df_scholar$title[w], 
                                       orcid_title=scores$cols[which_top_score])
scholar_to_orcid_matches
</code></pre>

<pre><code>##    top_score
## 1  0.9090909
## 2  1.0000000
## 3  0.2000000
## 4  0.9230769
## 5  0.2142857
## 6  0.9230769
## 7  0.8181818
## 8  0.9090909
## 9  0.8125000
## 10 0.9375000
## 11 1.0000000
## 12 1.0000000
## 13 0.9090909
## 14 0.8888889
## 15 0.0312500
## 16 0.8571429
## 17 1.0000000
## 18 0.9333333
## 19 0.8333333
## 20 1.0000000
## 21 0.0000000
## 22 0.9000000
## 23 0.9285714
## 24 1.0000000
## 25 0.8333333
## 26 1.0000000
## 27 0.0000000
## 28 0.9090909
## 29 1.0000000
## 30 0.0000000
## 31 0.0000000
## 32 0.0000000
## 33 0.0000000
## 34 0.0000000
## 35 0.0000000
## 36 0.0000000
## 37 0.0000000
## 38 0.0000000
## 39 0.0000000
## 40 0.0000000
## 41 0.0000000
## 42 0.0000000
## 43 0.0000000
## 44 0.0000000
## 45 0.0000000
## 46 0.0312500
## 47 1.0000000
## 48 0.6428571
## 49 1.0000000
## 50 0.0000000
##                                                                                                                                      scholar_title
## 1                                                               STRING v10: protein–protein interaction networks, integrated over the tree of life
## 2                                                      STRING v9. 1: protein-protein interaction networks, with increased coverage and integration
## 3                                                                  Comparative assessment of large-scale data sets of protein–protein interactions
## 4                                    The STRING database in 2017: quality-controlled protein–protein association networks, made broadly accessible
## 5                                      Sequence and comparative analysis of the chicken genome provide unique perspectives on vertebrate evolution
## 6                                                            STRING 8—a global view on proteins and their functional interactions in 630 organisms
## 7                                            STRING: known and predicted protein–protein associations, integrated and transferred across organisms
## 8                                                           STRING 7—recent developments in the integration and prediction of protein interactions
## 9                              The HUPO PSI&#39;s molecular interaction format—a community standard for the representation of protein interaction data
## 10  STRING v11: protein–protein association networks with increased coverage, supporting functional discovery in genome-wide experimental datasets
## 11                                                       eggNOG v3. 0: orthologous groups covering 1133 organisms at 41 different taxonomic ranges
## 12                                                                                  eggNOG v4. 0: nested orthology inference across 3686 organisms
## 13                                                        STITCH 5: augmenting protein–chemical interaction networks with tissue and affinity data
## 14                                                                           STITCH 4: integration of protein–chemical interactions with user data
## 15                                   Prion protein devoid of the octapeptide repeat region restores susceptibility to scrapie in PrP knockout mice
## 16                                                                                           STITCH 3: zooming in on protein–chemical interactions
## 17 eggNOG v2. 0: extending the evolutionary genealogy of genes with enhanced non-supervised orthologous groups, species and functional annotations
## 18                                        Version 4.0 of PaxDb: protein abundance data, integrated across model organisms, tissues, and cell‐lines
## 19                                     Millimeter‐scale genetic gradients and community‐level molecular convergence in a hypersaline microbial mat
## 20              MLTreeMap-accurate Maximum Likelihood placement of environmental DNA sequences into taxonomic and functional reference phylogenies
## 21                                                                           Distinct and regulated activities of human Gli proteins in Drosophila
## 22                                                      Shared components of protein complexes—versatile building blocks or biochemical artefacts?
## 23                                       A comprehensive set of protein complexes in yeast: mining large scale protein–protein interaction screens
## 24      Sequence-based factors influencing the expression of heterologous genes in the yeast Pichia pastoris--A comparative view on 79 human genes
## 25                                                                                          RAIN: RNA–protein Association and Interaction Networks
## 26                                                    Large-scale functional analysis of the roles of phosphorylation in yeast metabolic pathways.
## 27                                                                            Analyses of non-coding somatic drivers in 2,658 cancer whole genomes
## 28                                                                The SIB Swiss Institute of Bioinformatics’ resources: focus on curated databases
## 29                                                                              Viruses. STRING: A virus-host protein-protein interaction database
## 30                                                                       Effects of oral antibiotics and isotretinoin on the murine gut microbiota
## 31                                                The Helmholtz Network for Bioinformatics: an integrative web portal for bioinformatics resources
## 32                                               Cross-regulation between TDP-43 and paraspeckles promotes pluripotency-differentiation transition
## 33                                                                             Pathway and network analysis of more than 2500 whole cancer genomes
## 34               Analysis of the human kinome and phosphatome by mass cytometry reveals overexpression-induced effects on cancer-related signaling
## 35               Analysis of the Human Kinome and Phosphatome by Mass Cytometry Reveals Overexpression-Induced Effects on Cancer-Related Signaling
## 36                                            Protein tyrosine phosphatase non-receptor type 22 modulates colitis in a microbiota-dependent manner
## 37                                                                                         Mo1813 Antibiotics Provoke a Dysbiosis-Isotretinoin Not
## 38                                                                                                          ISMB 2014 Proceedings Papers Committee
## 39                                      Analysis of the human kinome and phosphatome reveals diseased signaling networks induced by overexpression
## 40                                                                   Sa1872 EBI2 Plays a Role in the Development of Intestinal Lymphoid Structures
## 41                                                                                                                                    eggNOG v3. 0
## 42                                                                                                                                    eggNOG v2. 0
## 43                                                              STRING and STITCH: known and predicted interactions between proteins and chemicals
## 44                                                                        Proteinkomplexe und Netzwerke: Neue Herausforderungen fuer die Proteomik
## 45                 Evolutionary Conservation of Hedgehog Signal Transduction Components-function of Vertebrate Proteins in Drosophila Melanogaster
## 46                                   Prion Protein Devoid of the Octapeptide Repeat Region Restores Susceptibility to Scrapie in PrP Knockout Mice
## 47                                                    MAPseq: highly efficient k-mer search with confidence estimates, for rRNA sequence analysis.
## 48                                                                              Proteome Analysis of Anopheles gambiae and Drosophila melanogaster
## 49                                                            Science Supporting Online Material Comparative Metagenomics of Microbial Communities
## 50                                                                                                      Pulling protein functions from metagenomes
##                                                                                                                                       orcid_title
## 1                                                              STRING v10: protein-protein interaction networks, integrated over the tree of life
## 2                                                      STRING v9.1: Protein-protein interaction networks, with increased coverage and integration
## 3                          Rapid Inference of Direct Interactions in Large-Scale Ecological Networks from Heterogeneous Microbial Sequencing Data
## 4                                   The STRING database in 2017: quality-controlled protein-protein association networks, made broadly accessible
## 5       Sequence-based factors influencing the expression of heterologous genes in the yeast Pichia pastoris-A comparative view on 79 human genes
## 6                                                         STRING 8 - A global view on proteins and their functional interactions in 630 organisms
## 7                                           STRING: Known and predicted protein-protein associations, integrated and transferred across organisms
## 8                                                        STRING 7 - Recent developments in the integration and prediction of protein interactions
## 9                           The HUPO PSI&#39;s Molecular Interaction format - A community standard for the representation of protein interaction data
## 10 STRING v11: protein-protein association networks with increased coverage, supporting functional discovery in genome-wide experimental datasets
## 11                                                       eggNOG v3.0: Orthologous groups covering 1133 organisms at 41 different taxonomic ranges
## 12                                                                                  eggNOG v4.0: nested orthology inference across 3686 organisms
## 13                                                       STITCH 5: augmenting protein-chemical interaction networks with tissue and affinity data
## 14                                                                          STITCH 4: integration of protein-chemical interactions with user data
## 15                                    Expression of amino-terminally truncated PrP in the mouse leading to ataxia and specific cerebellar lesions
## 16                                                                                          STITCH 3: Zooming in on protein-chemical interactions
## 17 eggNOG v2.0: Extending the evolutionary genealogy of genes with enhanced non-supervised orthologous groups, species and functional annotations
## 18                                       Version 4.0 of PaxDb: Protein abundance data, integrated across model organisms, tissues, and cell-lines
## 19                                    Millimeter-scale genetic gradients and community-level molecular convergence in a hypersaline microbial mat
## 20           MLTreeMap - accurate Maximum Likelihood placement of environmental DNA sequences into taxonomic and functional reference phylogenies
## 21                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 22                                                   Shared components of protein complexes - Versatile building blocks or biochemical artefacts?
## 23                                      A comprehensive set of protein complexes in yeast: Mining large scale protein-protein interaction screens
## 24      Sequence-based factors influencing the expression of heterologous genes in the yeast Pichia pastoris-A comparative view on 79 human genes
## 25                                                                                         RAIN: RNA-protein Association and Interaction Networks
## 26                                                    Large-scale functional analysis of the roles of phosphorylation in yeast metabolic pathways
## 27                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 28                                                               The SIB Swiss Institute of Bioinformatics&#39; resources: focus on curated databases
## 29                                                                              Viruses.STRING: A Virus-Host Protein-Protein Interaction Database
## 30                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 31                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 32                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 33                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 34                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 35                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 36                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 37                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 38                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 39                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 40                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 41                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 42                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 43                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 44                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 45                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
## 46                                    Expression of amino-terminally truncated PrP in the mouse leading to ataxia and specific cerebellar lesions
## 47                                                    MAPseq: highly efficient k-mer search with confidence estimates, for rRNA sequence analysis
## 48                                                      Comparative genome and proteome analysis of Anopheles gambiae and Drosophila melanogaster
## 49                                                                                              Comparative metagenomics of microbial communities
## 50                   Shotgun proteomics data from multiple organisms reveals remarkable quantitative conservation of the eukaryotic core proteome
</code></pre>

<pre><code class="r"># View(scholar_to_orcid_matches)
cutoff &lt;- .799
keep &lt;- top_score &gt; cutoff
df_scholar$probable_doi &lt;- NA
df_scholar$probable_doi[which(w)[keep]] &lt;- df_orcid$doi[which_top_score[keep]]

# found in Google Scholar, but unable to match to DOI/ORCID
with(df_scholar, title[is.na(doi) &amp; is.na(probable_doi)])
</code></pre>

<pre><code>##  [1] &quot;Comparative assessment of large-scale data sets of protein–protein interactions&quot;                                                  
##  [2] &quot;Sequence and comparative analysis of the chicken genome provide unique perspectives on vertebrate evolution&quot;                      
##  [3] &quot;Prion protein devoid of the octapeptide repeat region restores susceptibility to scrapie in PrP knockout mice&quot;                    
##  [4] &quot;Distinct and regulated activities of human Gli proteins in Drosophila&quot;                                                            
##  [5] &quot;Analyses of non-coding somatic drivers in 2,658 cancer whole genomes&quot;                                                             
##  [6] &quot;Effects of oral antibiotics and isotretinoin on the murine gut microbiota&quot;                                                        
##  [7] &quot;The Helmholtz Network for Bioinformatics: an integrative web portal for bioinformatics resources&quot;                                 
##  [8] &quot;Cross-regulation between TDP-43 and paraspeckles promotes pluripotency-differentiation transition&quot;                                
##  [9] &quot;Pathway and network analysis of more than 2500 whole cancer genomes&quot;                                                              
## [10] &quot;Analysis of the human kinome and phosphatome by mass cytometry reveals overexpression-induced effects on cancer-related signaling&quot;
## [11] &quot;Analysis of the Human Kinome and Phosphatome by Mass Cytometry Reveals Overexpression-Induced Effects on Cancer-Related Signaling&quot;
## [12] &quot;Protein tyrosine phosphatase non-receptor type 22 modulates colitis in a microbiota-dependent manner&quot;                             
## [13] &quot;Mo1813 Antibiotics Provoke a Dysbiosis-Isotretinoin Not&quot;                                                                          
## [14] &quot;ISMB 2014 Proceedings Papers Committee&quot;                                                                                           
## [15] &quot;Analysis of the human kinome and phosphatome reveals diseased signaling networks induced by overexpression&quot;                       
## [16] &quot;Sa1872 EBI2 Plays a Role in the Development of Intestinal Lymphoid Structures&quot;                                                    
## [17] &quot;eggNOG v3. 0&quot;                                                                                                                     
## [18] &quot;eggNOG v2. 0&quot;                                                                                                                     
## [19] &quot;STRING and STITCH: known and predicted interactions between proteins and chemicals&quot;                                               
## [20] &quot;Proteinkomplexe und Netzwerke: Neue Herausforderungen fuer die Proteomik&quot;                                                         
## [21] &quot;Evolutionary Conservation of Hedgehog Signal Transduction Components-function of Vertebrate Proteins in Drosophila Melanogaster&quot;  
## [22] &quot;Prion Protein Devoid of the Octapeptide Repeat Region Restores Susceptibility to Scrapie in PrP Knockout Mice&quot;                    
## [23] &quot;Proteome Analysis of Anopheles gambiae and Drosophila melanogaster&quot;                                                               
## [24] &quot;Pulling protein functions from metagenomes&quot;
</code></pre>

<pre><code class="r">write.table(df_scholar, scholar_file, sep=&quot;\t&quot;, quote=FALSE, row.names = FALSE)



# what is in ORCID, but not in Google Scholar?
scholar_dois &lt;- unname(apply(df_scholar[,c(&quot;doi&quot;,&quot;probable_doi&quot;)], 
                             1, function(u) na.omit(u)[1]))
m &lt;- match(df_orcid$doi, scholar_dois)
df_orcid$scholar_id &lt;- df_scholar$pubid[m]

df_orcid %&gt;% filter(is.na(scholar_id))
</code></pre>

<pre><code>##                                                                                                            title
## 1 Disentangling the impact of environmental and phylogenetic constraints on prokaryotic within-species diversity
## 2                                                         RAIN: RNA-protein Association and Interaction Networks
## 3                                               A Family of Interaction-Adjusted Indices of Community Similarity
## 4                            MAPseq: improved speed, accuracy and consistency in ribosomal RNA sequence analysis
##                                                       journal            type                       doi scholar_id
## 1                                            The ISME Journal journal-article 10.1038/s41396-020-0600-z       &lt;NA&gt;
## 2 Database : the journal of biological databases and curation journal-article   10.1093/database/baw167       &lt;NA&gt;
## 3                                                        &lt;NA&gt;           other            10.1101/040097       &lt;NA&gt;
## 4                                                        &lt;NA&gt;           other            10.1101/126953       &lt;NA&gt;
</code></pre>

<pre><code class="r">write.table(df_orcid, orcid_file, sep=&quot;\t&quot;, quote=FALSE, row.names = FALSE)


# my_search &lt;- &quot;trypsinosome&quot;
# grep(my_search, df_scholar$title[w], value=TRUE)
# grep(my_search, df_orcid$title, value=TRUE)
</code></pre>

</body>

</html>
